src/test/java/org/broadinstitute/hellbender/engine/GATKToolUnitTest.java

```diff
@@ -191,4 +201,82 @@ public void testReadsValidationStringencySilent(final String bamFileName, final
         testValidationStringency(bamFileName, referenceFileName, "SILENT", count);
     }
 
+    @Test(expectedExceptions = UserException.IncompatibleSequenceDictionaries.class)
+    public void testDisallowLexicographicallySortedVariantHeader_ifClashWithReference() throws Exception {
```

Comment:
Move this test case next to `testAllowLexicographicallySortedVariantHeader()` in this file to improve organization.